connector/src/main/java/com/google/cloud/hive/bigquery/connector/input/BigQueryFilters.java

```diff
@@ -24,56 +24,100 @@
-      function.setGenericUDF(new BigQueryUDFDateSub());
-    } else if (function.getGenericUDF() instanceof GenericUDFDateAdd) {
-      function.setGenericUDF(new BigQueryUDFDateAdd());
-    } else if (function.getGenericUDF() instanceof GenericUDFOPMod) {
-      function.setGenericUDF(new BigQueryUDFMod());
-    } else if (function.getGenericUDF() instanceof GenericUDFRegExp) {
-      function.setGenericUDF(new BigQueryUDFRegExpContains());
-    } else if (function.getGenericUDF() instanceof GenericUDFToDate) {
-      function.setGenericUDF(new BigQueryUDFToDate());
-    } else if (function.getGenericUDF() instanceof GenericUDFTimestamp) {
-      function.setGenericUDF(new BigQueryUDFToDatetime());
-    } else if (function.getGenericUDF() instanceof GenericUDFToTimestampLocalTZ) {
-      function.setGenericUDF(new BigQueryUDFToTimestamp());
-    } else if (function.getGenericUDF() instanceof GenericUDFToBinary) {
-      function.setGenericUDF(new BigQueryUDFToBytes());
-    } else if (function.getGenericUDF() instanceof GenericUDFToVarchar) {
-      function.setGenericUDF(new BigQueryUDFToString());
-    } else if (function.getGenericUDF() instanceof GenericUDFToChar) {
-      function.setGenericUDF(new BigQueryUDFToString());
-    } else if (function.getGenericUDF() instanceof GenericUDFToDecimal) {
-      function.setGenericUDF(new BigQueryUDFToDecimal());
-    } else if (function.getGenericUDF() instanceof GenericUDFBridge) {
-      switch (function.getGenericUDF().getUdfName()) {
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryFilters.class);
+
+  // List of UDFs and operators that are exactly the same in Hive and BigQuery
+  // TODO: Make sure we make this list as comprehensive as possible
+  protected static List<String> identicalUDFs;
+
+  static {
+    identicalUDFs = new ArrayList<>();
+    for (String udf :
+        new String[] {
+          "GenericUDFOPEqual",
+          "GenericUDFOPNotEqual",
+          "GenericUDFOPGreaterThan",
+          "GenericUDFOPLessThan",
+          "GenericUDFOPEqualOrGreaterThan",
+          "GenericUDFOPEqualOrLessThan",
+          "GenericUDFIn",
+          "GenericUDFBetween",
+          "GenericUDFOPNot",
+          "GenericUDFOPNull",
+          "GenericUDFOPNotNull",
+          "GenericUDFOPTrue",
+          "GenericUDFOPNotTrue",
+          "GenericUDFOPFalse",
+          "GenericUDFOPNotFalse",
+          "GenericUDFOPAnd",
+          "GenericUDFOPOr",
+          "GenericUDFOPPlus",
+          "GenericUDFOPMinus",
+          "GenericUDFOPTrue",
+          "GenericUDFOPFalse",
+          "GenericUDFOPNegative",
+          "GenericUDFOPPositive",
+          "GenericUDFOPPower",
+          "GenericUDFOPDivide",
+          "GenericUDFOPMultiply",
+        }) {
+      identicalUDFs.add("org.apache.hadoop.hive.ql.udf.generic." + udf);
+    }
+  }
+
+  /** Converts the Hive UDF to the corresponding BigQuery function */
+  protected static GenericUDF convertUDF(GenericUDF udf) {
```

Comment:
wonder if we can gather these into a map of (className -> className) and utilize hive or hadoop's ReflectionUtil.newInstance

https://github.com/apache/hive/blob/master/common/src/java/org/apache/hive/common/util/ReflectionUtil.java#L74

Reply:
I've changed it to use classes instead of strings, so it'll be less error-prone. I don't think we need to generate instances in this cases as we just need to compare the UDF types.

Reply:
i mean for the list of non-identical udfs. 

(udf instanceof GenericUDFToBinary) {
      return new BigQueryUDFToBytes();
}

can be something udf classname mapping to bigquery udf classname and use reflection.

it just a coding choice, either way is fine.

Reply:
on 2nd thought, maybe good to keep this way, less dependency on hive stuff.